src/rt-app.c

```diff
@@ -415,6 +434,72 @@ static int run_event(event_data_t *event, int dry_run,
 			pthread_yield();
 		}
 		break;
+	case rtapp_fork:
+		{
+			log_debug("fork");
+
+			/*
+			 * If multiple threads race to fork, we must ensure
+			 * each one sees a unique index. Hence the lock.
+			 */
+			pthread_mutex_lock(&fork_mutex);
+			int new_thread_index = nthreads++;
+			threads = realloc(threads, nthreads * sizeof(*threads));
+			thread_data_t *tdata = malloc(sizeof(thread_data_t));
+
+			if (!threads || !tdata) {
+				log_error("Failed to allocate memory for a new fork: %s", rdata->res.fork.tdata->name);
+				pthread_mutex_unlock(&fork_mutex);
+				exit(EXIT_FAILURE);
+			}
+
+			/* Find the thread data associated with this fork */
+			if (!rdata->res.fork.tdata) {
+				rdata->res.fork.tdata = find_thread_data(rdata->res.fork.ref, &opts);
+			}
+
+			/*
+			 * We have one tdata created at config parse, but we
+			 * might spawn multiple threads if we were running in
+			 * a loop, so ensure we duplicate the tdata before
+			 * creating each thread, and we should free it at the
+			 * end of the thread_body()
+			 */
+			memcpy(tdata, rdata->res.fork.tdata, sizeof(*tdata));
+
+			/* Mark this thread as forked */
+			tdata->forked = 1;
+			/* update the index value */
+			tdata->ind = new_thread_index;
+
+			/* Make sure each forked thread has a unique name */
+			int string_size = strlen(tdata->name) + 1 /* NULL */ + 10 /* postfix */;
+			char *unique_name = malloc(string_size);
+			if (unique_name) {
+				snprintf(unique_name, string_size, "%s%03d", tdata->name, rdata->res.fork.nforks++);
```

Comment:
this is different from default naming convention.
I mean if we have "instance" :2, the name of the tasks will be the same for the 2 tasks
but if we use "instance" :0 and then call twice "fork" : "taskname", the name of the 2 tasks will be different. We have to be consistent with both case. At now, we use data->ind which is unique to make difference between the task.
Futhermore, if I describe 11 tasks in my json file form "task1" to "task11" and then I fork "task1", i will have 2 tasks with same name "tasks11" ... So you are not unique 
I understand the interest of uniqueness but for now data->ind ensure it. I fyou want unique name, you needs more complex mecanism and you will have to extend it to tasks create at beg of the use case and not only for forked task

Reply:
I agree about the inconsistencies in behavior if we use "instance": 2.

> Futhermore, if I describe 11 tasks in my json file form "task1" to "task11" and then I fork "task1", i will have 2 tasks with same name "tasks11

This is not true as we use "%s03d" so we will have task1001 not task11. But yes we are not guaranteed uniqueness if something called a task1001.

So the proposal is to always append data->ind to the task name? The advantage of nforks used now is that it'll increment each fork independently.

for example if you have 2 fork events in a task; `"fork": "thread1"` and `"fork": "thread2"` each fork will start appending a 'unique' number starting from 000. This will be very helpful when looking at traces and trying to identify a particular run. Using ind there will be a bit of homework to be done to find out what's the name of the 8th fork of a particular event for example.

Maybe we can have a pattern like this: `"%s-%d-%04d", name, ind, nforks`?

Any thing forked at the beginning will only have `"%s-%d", name, ind` only